server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java

```diff
@@ -332,4 +332,22 @@ public int compareTo(ByteSizeValue other) {
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         return builder.value(toString());
     }
+
+    /**
+     * @return Constructs a {@link ByteSizeValue} with the bytes resulting from the addition of the arguments' bytes. Note that the
+     *         resulting {@link ByteSizeUnit} is bytes.
+     * @throws ArithmeticException if the resulting bytes overflow a long
```

Comment:
While true, it seems unncessary to document it? Is a runtime exception and should never happen.

Reply:
Sure, removed the documentation line for this overflow exception.